### PR TITLE
Mf 159 training provider search

### PR DESCRIPTION
**Pull request by robinoconnor**

### CODE REVIEW

File: src/SFA.DAS.Reservations.Api/StartupExtensions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -19,5 +23,14 @@ public static IServiceCollection AddNServiceBusClientUnitOfWork(this IServiceCol
                 .AddScoped<IUnitOfWork, UnitOfWork.NServiceBus.Features.ClientOutbox.Pipeline.UnitOfWork>()
                 .AddScoped<IUnitOfWorkManager, UnitOfWorkManager>();
         }
+
+        public static void AddElasticSearch(this IServiceCollection collection, ReservationsConfiguration configuration)
+        {
+            var connectionPool = new  SingleNodeConnectionPool(new Uri(configuration.ElasticSearchServerUrl));
```
Comment (dashton82) [inline src/SFA.DAS.Reservations.Api/StartupExtensions/ServiceCollectionExtensions.cs]: Does this need to connect differently? Using the basic auth?
```suggestion
            var connectionPool = new  SingleNodeConnectionPool(new Uri(configuration.ElasticSearchServerUrl));
            
            var settings = new ConnectionSettings(connectionPool);
            
            if (!string.IsNullOrEmpty(configuration.ElasticSearchUsername) &&
                !string.IsNullOrEmpty(configuration.ElasticSearchPassword))
            {
                settings.BasicAuthentication(configuration.ElasticSearchUsername, configuration.ElasticSearchPassword);
            }
            

            collection.AddTransient<IElasticClient>(sp => new ElasticClient(settings));
```

File: src/SFA.DAS.Reservations.Data/Repository/ReservationIndexRepository.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+        public ReservationIndexRepository(IElasticClient client, ReservationsApiEnvironment environment)
+        {
+            _client = client;
+            _environment = environment;
+        }
+
+        public async Task<IEnumerable<ReservationIndex>> Find(long providerId, string term)
+        {
+            var searchIndexRegistryResponse = _client.Search<IndexRegistryEntry>(s => s
```
Comment (dashton82) [inline src/SFA.DAS.Reservations.Data/Repository/ReservationIndexRepository.cs]: Can this be SearchAsync? Additional enhancement to note here might be to cache that reservationIndexName for a day

**Comment by dashton82** [comment]
All good 👍 just a couple of queries

File: src/SFA.DAS.Reservations.Api/StartupExtensions/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (robinoconnor) [reply src/SFA.DAS.Reservations.Api/StartupExtensions/ServiceCollectionExtensions.cs]: Yes at the time i wrote that part it was clear how we were going to connect to the AT and beyond environments and I missed updated this afterwards so will change this now

File: src/SFA.DAS.Reservations.Data/Repository/ReservationIndexRepository.cs (hunk 2 above)
Comment (robinoconnor) [reply src/SFA.DAS.Reservations.Data/Repository/ReservationIndexRepository.cs]: yep good point on the caching, i'll take a look how that can be implemented.  With regards to the async I have a feeling I had trouble with it before but i'll double check and see if I can make the change. May have just been not working due to something else.

File: src/SFA.DAS.Reservations.Api/StartupExtensions/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (dashton82) [reply src/SFA.DAS.Reservations.Api/StartupExtensions/ServiceCollectionExtensions.cs]: 💥

File: src/SFA.DAS.Reservations.Data/Repository/ReservationIndexRepository.cs (hunk 2 above)
Comment (dashton82) [reply src/SFA.DAS.Reservations.Data/Repository/ReservationIndexRepository.cs]: Can leave the caching for now 😄

Comment (robinoconnor) [reply src/SFA.DAS.Reservations.Data/Repository/ReservationIndexRepository.cs]: ok shall we add a story for it in the backlog?

**Pull request merged**